lib/Db/MessageMapper.php

```diff
@@ -62,6 +62,23 @@ public function findAllUids(Mailbox $mailbox): array {
 		return $uids;
 	}
 
+	public function findHighestUid(Mailbox $mailbox): ?int {
+		$query = $this->db->getQueryBuilder();
+
+		$query->select($query->createFunction('MAX(' . $query->getColumnName('uid') .  ')'))
```

Comment:
max is available in 18+,
maybe leave yourself a TODO?

Reply:
Yes I used the function until I saw the failure on stable17 :)